### PR TITLE
fix: assert failure on non-policy asset consolidation in CreateTransactionInternal 

### PR DESCRIPTION
**Pull request by delta1**
same diff as #1258 but for `master` branch

### CODE REVIEW

**Comment by jamesdorfman** [comment]
@delta1 there's a test failure in `[ASan + LSan + UBSan + integer, no depends] [jammy]` in the new regression test added here.

```
20/246 - 
wallet_elements_regression_1259.py --legacy-wallet
 failed, Duration: 1 s
stdout:
2023-10-18T16:32:27.728000Z TestFramework (INFO): Initializing test directory /tmp/cirrus-ci-build/ci/scratch/test_runner/test_runner_₿_🏃_20231018_163042/wallet_elements_regression_1259_223
2023-10-18T16:32:28.615000Z TestFramework (ERROR): Unexpected exception caught during testing
Traceback (most recent call last):
  File "/tmp/cirrus-ci-build/ci/scratch/build/elements-x86_64-pc-linux-gnu/test/functional/test_framework/test_framework.py", line 132, in main
    self.run_test()
  File "/tmp/cirrus-ci-build/ci/scratch/build/elements-x86_64-pc-linux-gnu/test/functional/wallet_elements_regression_1259.py", line 38, in run_test
    self.nodes[0].generate(COINBASE_MATURITY + 1)
  File "/tmp/cirrus-ci-build/ci/scratch/build/elements-x86_64-pc-linux-gnu/test/functional/test_framework/test_node.py", line 317, in generate
    return self.generatetoaddress(nblocks=nblocks, address=self.get_deterministic_priv_key().address, maxtries=maxtries, **kwargs)
TypeError: TestNode.generatetoaddress() missing 1 required keyword-only argument: 'invalid_call'
```

**Comment by delta1** [comment]
Pushed the fix to the functional test for the `generate` calls, which has exposed a deeper issue in the changes since the 22 branch: 

```
AssertionError: Unexpected stderr elementsd: wallet/spend.cpp:1272: bool wallet::CreateTransactionInternal(wallet::CWallet&, const std::vector<wallet::CRecipient>&, CTransactionRef&, CAmount&, int&, bilingual_str&, const wallet::CCoinControl&, FeeCalculation&, bool, wallet::BlindDetails*, const wallet::IssuanceDetails*): Assertion `change_and_fee >= 0' failed. 
```

**Comment by delta1** [comment]
Thanks @jamesdorfman, testing now after fixing a bug I introduced in the test.

**Comment by delta1** [comment]
Pushed to fix lint issues

**Comment by delta1** [comment]
tACK 5b72ae23afcb9a2d21a3ce7e1cfc40e104e9607b

**Comment by delta1** [comment]
ACK 04cba29faf6383d99a47d373c58cc52eacd3a4ad

File: test/functional/example_elements_code_tutorial.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+class WalletTest(BitcoinTestFramework):
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.num_nodes = 2
+        self.extra_args = [[
+            "-blindedaddresses=1",
+            "-initialfreecoins=2100000000000000",
+            "-con_blocksubsidy=0",
+            "-con_connect_genesis_outputs=1",
+            "-txindex=1",
```
Comment (delta1) [inline test/functional/example_elements_code_tutorial.py:22]: I think these args should match the conf files from the tutorial

File: test/functional/example_elements_code_tutorial.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        assert issuances[0]['isreissuance'] or issuances[1]['isreissuance']
+
+        self.generate(self.nodes[0], 1)
+
+        expected_amt = {
+            'bitcoin': 0,
+            '8f1560e209f6bcac318569a935a0b2513c54f326ee4820ccd5b8c1b1b4632373': 0,
+            '4fa41f2929d4bf6975a55967d9da5b650b6b9bfddeae4d7b54b04394be328f7f': 99
+        }
+        assert self.nodes[0].gettransaction(reissuance_txid)['amount'] == expected_amt
```
Comment (delta1) [inline test/functional/example_elements_code_tutorial.py:70]: I think we could use some asserts of the balances before and after this whole process. not a showstopper though.

File: test/functional/test_runner.py [hunk 3]
```diff
@@ -88,6 +88,7 @@
 BASE_SCRIPTS = [
     # Scripts that are run by default.
     # vv First elements tests vv
+    'example_elements_code_tutorial.py'
```
Comment (delta1) [inline test/functional/test_runner.py]:
```suggestion
    'example_elements_code_tutorial.py,'
```

Comment (delta1) [reply test/functional/test_runner.py]: addressed this in the fixup commit

**Review by delta1 — COMMENTED** [review]
tACK 406ce9a42c3b271d084fe2a42029f6cad3c9b54e 

ran unit and functional tests successfully

**Pull request merged**